sklearn/inspection/_partial_dependence.py

```diff
@@ -954,3 +1111,470 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
+          ...                   plot_individual_conditional_expectation)
+          >>> from sklearn.datasets import make_friedman1
+          >>> from sklearn.linear_model import LinearRegression
+          >>> X, y = make_friedman1()
+          >>> est = LinearRegression().fit(X, y)
+          >>> disp1 = plot_individual_conditional_expectation(
+          ...     est, X)  # doctest: +SKIP
+          >>> disp2 = plot_individual_conditional_expectation(
+          ...     est, X,ax=disp1.axes_)  # doctest: +SKIP
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    estimator : BaseEstimator
+        A fitted estimator object implementing :term:`predict`,
+        :term:`predict_proba`, or :term:`decision_function`.
+        Multioutput-multiclass classifiers are not supported.
+
+    X : {array-like or dataframe} of shape (n_samples, n_features)
+        The data to use to build the grid of values on which the individual
+        conditional expectation will be evaluated.
+        This is usually the training data.
+
+    features : list of {int, str}
+        The target features for which to create the ICEs.
+        if any entry is a string, then it must be in ``feature_names``.
+
+    response_method : 'auto', 'predict_proba' or 'decision_function', \
+            optional (default='auto')
+        Specifies whether to use :term:`predict_proba` or
+        :term:`decision_function` as the target response. For regressors
+        this parameter is ignored and the response is always the output of
+        :term:`predict`. By default, :term:`predict_proba` is tried first
+        and we revert to :term:`decision_function` if it doesn't exist.
+
+    percentiles : tuple of float, optional (default=(0.05, 0.95))
+        The lower and upper percentile used to create the extreme values
+        for the ICE axes. Must be in [0, 1].
+
+    grid_resolution : int, optional (default=100)
+        The number of equally spaced points on the axes of the plots, for each
+        target feature.
+
+    fixed_start_point : bool, optional (default=True)
+        Whether to use a fixed starting point for all the ICE curves.
+
+    feature_names : array-like of shape (n_features,), dtype=str, default=None
+        Name of each feature; feature_names[i] holds the name of the feature
+        with index i.
+        By default, the name of the feature corresponds to their numerical
+        index for NumPy array and their column name for pandas dataframe.
+
+    target : int, optional (default=None)
+        - In a multiclass setting, specifies the class for which the ICEs
+          should be computed. Note that for binary classification, the
+          positive class (index 1) is always used.
+        - In a multioutput setting, specifies the task for which the ICEs
+          should be computed.
+
+        Ignored in binary classification or classical regression settings.
+
+    n_cols : int, optional (default=3)
```

Comment:
Basically remove all the optional occurrences in your added docstring

Reply:
https://github.com/scikit-learn/scikit-learn/pull/16164/commits/ba8b58bfddaa65c749eeadbc5d47edf0e01c0ac7